Gems/ROS2/Code/Include/ROS2/Manipulator/MotorizedJointComponent.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "MotorizedJointBus.h"
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Math/Transform.h>
+#include <AzCore/Math/Vector2.h>
+#include <ROS2/VehicleDynamics/DriveModels/PidConfiguration.h>
+
+namespace ROS2
+{
+    //! A prototype component for simulated joint with a motor.
+    //! It works with either TransformBus or RigidBodyBus.
+    //! TransformBus mode, called `AnimationMode` changes local transform. In this mode, you cannot have a rigid body
+    //! controller enabled. With RigidBodyBus it applies forces and torque according to PID control.
+    //! @note This class is already used through ROS2FrameComponent.
+    class MotorizedJointComponent
+        : public AZ::Component
+        , public AZ::TickBus::Handler
+        , public MotorizedJointRequestBus::Handler
+    {
+    public:
+        AZ_COMPONENT(MotorizedJointComponent, "{AE9207DB-5B7E-4F70-A7DD-C4EAD8DD9403}", AZ::Component);
+
+        MotorizedJointComponent() = default;
+        ~MotorizedJointComponent() = default;
+        //////////////////////////////////////////////////////////////////////////
+        // Component overrides
+        void Activate() override;
+        void Deactivate() override;
+        static void Reflect(AZ::ReflectContext* context);
+        //////////////////////////////////////////////////////////////////////////
+
+        ////////////////////////////////////////////////////////////////////////
+        // MotorizedJointRequestBus::Handler overrides
+        void SetSetpoint(float setpoint) override;
+        float GetSetpoint() override;
+        float GetError() override;
+        float GetCurrentMeasurement() override;
+        ////////////////////////////////////////////////////////////////////////
+
+        //! Get a degree of freedom direction.
+        //! @returns direction of joint movement in global coordinates.
+        AZ::Vector3 GetDir() const
+        {
+            return m_jointDir;
+        };
+
+    private:
+        float ComputeMeasurement(AZ::ScriptTimePoint time);
+        void SetVelocity(float velocity, float deltaTime);
+        void ApplyLinVelAnimation(float velocity, float deltaTime);
+        void ApplyLinVelRigidBodyImpulse(float velocity, float deltaTime);
+        void ApplyLinVelRigidBody(float velocity, float deltaTime);
```

Comment:
Can we add in-line comments to describe what units ```velocity``` and ```deltaTime``` are expected to be in?
e.g. 
```
void SetVelocity(float velocity /* m/s */, float deltaTime /* microseconds */);
```

Reply:
done